src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -94,10 +106,35 @@ void resolverEstimate() {
     void writeRoundState() { }
 
     /** Gather statistical data of the current round and update the aggregated simulation statistics based on all rounds. */
-    void updateSimulationStats() { }
+    void updateSimulationStats() {
+        Map<Beacon, Location> beaconsToEstimatedLocations = ((GlobalResolver) resolver).getBeaconsToEstimatedLocations();
+        List<Double> distances = beaconsToEstimatedLocations.keySet().stream() // ignore beacons that have never been observed
+                .map(beacon -> distance(beaconsToEstimatedLocations.get(beacon), beacon.getLocation())).collect(toList());
+
+        if (!distances.isEmpty()) { // distances is empty until the first round an observer observed a beacon
+            double min = distances.stream().min(Double::compareTo).get();
+            distancesStats.put("min", Math.min(distancesStats.getOrDefault("min", Double.POSITIVE_INFINITY), min));
```

Comment:
Consider using `Map.merge()` here (and in the following few rows) to simplify the code.